### ISSUE TITLE
ResizeObserver loop limit exceeded

### ISSUE THREAD

**Issue by dschwank** [asyncapi/asyncapi-react#498]
**Description**
Within the new `next` version we see some uncaught `ResizeObserver loop limit exceeded` exceptions (especially in E2E tests) as soon as the width is smaller than `1280px`. It seems to be related to this snippets here:
- https://github.com/asyncapi/asyncapi-react/blob/next/library/src/containers/AsyncApi/Layout.tsx#L30-L36
- https://github.com/asyncapi/asyncapi-react/blob/next/library/src/containers/AsyncApi/Layout.tsx#L44

We assume that the `ResizeObserver` is not able to deliver all observations within one animation frame due to the nested `setState()`. At least this comment leads to that assumption: https://stackoverflow.com/a/50387233/11471068

**Expected result**

- the AsyncAPI library does not cause any `ResizeObserver loop limit exceeded` exceptions

**Actual result**

- the AsyncAPI causes from time to time `ResizeObserver loop limit exceeded` exceptions

**Steps to reproduce**

1. Start the Playground on the `next` branch with `npm start`
2. Open the playground at `localhost:3000`
3. Open the devTools
4. Add the following listener to console

```js
window.addEventListener('error', (e) => {
  console.log('[UNCAUGHT]', e);
});
```

5. Resize the playground / code section, thus the panel is shown on the right side

Result: Now you should see a `ResizeObserver loop limit exceeded` in your console

**Troubleshooting**

This issue is not critical in the browser, but it causes our Cypress E2E tests to fail. As mitigation we ignore those uncaught exceptions:

```ts
// credits go to https://github.com/quasarframework/quasar/issues/2233#issuecomment-656464466

Cypress.on('uncaught:exception', (err) => {
  if (err.message.includes('ResizeObserver')) {
    // returning false here prevents Cypress from
    // failing the test
    return false;
  }
  return true;
});
```

**Comment by magicmatatjahu** [comment]
@dschwank Hello! Thanks for the issue. I created PR with possible fix (https://github.com/asyncapi/asyncapi-react/pull/499), but I need to reproduce your problem from the upstream version of component to check that everything is good, but tbh my computer is probably to fast and I don't see any exception from `ResizeObserver` 😅

**Comment by dschwank** [comment]
Hey @magicmatatjahu!

Thanks for your response and for waiting so long. I read your message and tested it right after you posted it, but somehow I missed to answer. Sorry for that!

The problem still exists with your fix - it's already better, as the re-render doesn't happen on each `width` change, but it still persists if the size changes between bigger / smaller `1280px`.

> but tbh my computer is probably to fast and I don't see any exception from ResizeObserver

We also only noticed it in our pipeline, as the exception is not printed. You could use the following snippet in your dev tools to make the error visible:

```js
window.addEventListener('error', (e) => {
  console.log('[UNCAUGHT]', e);
});
```

Somehow the `setState` takes quite some time, which blocks the resizeObserver. So far I haven't found a really nice solution. I just mitigated it by adding a `setTimeout(() => setState() , 0)` to your PR - then it works fine for me.

**Comment by magicmatatjahu** [comment]
@dschwank Hello! Thanks for suggestion and for snippet to show error! In mentioned  PR I wrapped the ResizeObserver callback with `requestAnimationFrame` to to have one execution per frame, and as I see it fixed the problem. I think that's better than `setTimeout(() => setState() , 0)`, of course `setState` will be called at the end of event queue, but that error is not related with setting state, but exactly with often execution of callback :) As I see you can test solution in your local env, so (if you have time) please check my latest change in PR and we will merge it :)

**Comment by dschwank** [comment]
Thanks a lot for the fast response. I just tested it - `requestAnimationFrame` would be fine for me 🥳

**Comment by magicmatatjahu** [comment]
Fixed in the 1.0.0-next.26

**Comment by dschwank** [comment]
Thanks a lot for fixing the issue!
